src/librustdoc/clean/mod.rs

```diff
@@ -1442,12 +1442,12 @@ impl<'tcx> Clean<Type> for Ty<'tcx> {
                 };
                 inline::record_extern_fqn(cx, did, kind);
                 let path = external_path(cx, did, false, vec![], substs);
-                ResolvedPath { path, did, is_generic: false }
```

Comment:
This path is for a `struct`/`enum`/`union` declaration, so its `Res` should be correct and preserve the `is_generic: false` behavior.